src/components/CompareResults/beta/DownloadButton.tsx

```diff
@@ -13,10 +16,76 @@ const styles = {
   }),
 };
 
+interface GroupedResults {
+  [key: string]: ResultObject[];
+}
+
+interface ResultObject {
+  header_name: string;
+  new_rev: string;
+}
+
 function DownloadButton() {
+  let fileName = 'perf-compare-all-revisions.json';
+  const results = useAppSelector((state) => {
+    if (
+      state.comparison.activeComparison ===
+      Strings.components.comparisonRevisionDropdown.allRevisions
+    ) {
+      return state.compareResults.data;
+    } else {
+      fileName = `perf-compare-${truncateHash(
+        state.comparison.activeComparison,
+      )}.json`;
```

Comment:
changing the filename inside the selector doesn't sound right. Selectors shouldn't have side effects.
Instead you can make the fileName the result of another `useAppSelector` call.